rust/src/interpreter/stack.rs

```diff
@@ -0,0 +1,149 @@
+use evmc_vm::{StatusCode, StepStatusCode};
+
+use crate::types::u256;
+
+#[derive(Debug)]
+pub struct Stack(Vec<u256>);
+
+impl Stack {
+    pub fn from(inner: Vec<u256>) -> Self {
+        Self(inner)
+    }
+
+    pub fn empty() -> Self {
+        Self(Vec::new())
+    }
+
+    pub fn into_inner(self) -> Vec<u256> {
+        self.0
+    }
+
+    pub fn push(&mut self, value: impl Into<u256>) -> Result<(), (StepStatusCode, StatusCode)> {
+        self.check_overflow_on_push()?;
+        self.0.push(value.into());
+        Ok(())
+    }
+
+    pub fn swap_with_top(&mut self, nth: usize) -> Result<(), (StepStatusCode, StatusCode)> {
+        self.check_underflow(nth + 1)?;
+
+        let len = self.0.len();
+        self.0.swap(len - 1, len - 1 - nth);
+        Ok(())
+    }
+
+    pub fn pop<const N: usize>(&mut self) -> Result<[u256; N], (StepStatusCode, StatusCode)> {
```

Comment:
Integral generics is a new feature I never got to use before. Happy that it is here now.